tests/plugins/test_docker_api.py

```diff
@@ -133,7 +133,7 @@ def raise_exc(*args, **kwargs):
         raise docker.errors.APIError(message='foo',
                                      response=http_error,
                                      explanation=explanation)
-        yield {}
+        yield {}    # pylint: disable=unreachable; this needs to be a generator, TODO: why?
```

Comment:
Because we want the error to be raised on the first *iteration* of the result, not when the function is first called.

Reply:
Ok, that makes sense. Moving the `yield` above the `raise` would then be incorrect, right?

Reply:
I don't think that would work, we need first iteration to raise the exception

Reply:
Thing is, the tests still work with the `yield` above the `raise`, but that might be a weird coincidence, I think I will keep it as is to be sure.